app/code/Bulbulatory/Recomendations/Controller/Recommendations/SaveRecommendation.php

```diff
@@ -0,0 +1,88 @@
+<?php
+namespace Bulbulatory\Recomendations\Controller\Recommendations;
+
+use Magento\Framework\Controller\ResultFactory;
+use Bulbulatory\Recomendations\Api\RecommendationRepositoryInterface;
+use Magento\Customer\Model\Session;
+use Magento\Backend\App\Action\Context;
+use Magento\Framework\Math\Random;
+use Bulbulatory\Recomendations\Api\Data\RecommendationInterface;
+use Bulbulatory\Recomendations\Helper\Email;
+use Magento\Framework\UrlInterface;
+use Magento\Customer\Api\CustomerRepositoryInterface;
+
+class SaveRecommendation extends \Magento\Framework\App\Action\Action
+{
+    const CONFIRMATION_URL = 'bulbulatory_recomendations/recommendations/confirmRecommendation';
+
+    protected $recommendationRepository;
+    protected $loggedCustomer;
+    protected $mathRandom;
+    protected $recommendation;
+    protected $email;
+    protected $urlBuilder;
+    protected $customerRepository;
+
+    public function __construct(
+        Context $context,
+        RecommendationRepositoryInterface $recommendationRepository,
+        Session $loggedCustomer,
+        Random $mathRandom,
+        RecommendationInterface $recommendation,
+        Email $email,
+        UrlInterface $urlBuilder,
+        CustomerRepositoryInterface $customerRepository
+    ) {
+        parent::__construct($context);
+        $this->recommendationRepository = $recommendationRepository;
+        $this->loggedCustomer = $loggedCustomer;
+        $this->mathRandom = $mathRandom;
+        $this->recommendation = $recommendation;
+        $this->email = $email;
+        $this->urlBuilder = $urlBuilder;
+        $this->customerRepository = $customerRepository;
+    }
+
+	public function execute()
+	{
+        $post = (array)$this->getRequest()->getPost();
+        if (!empty($post)) {
+            $email = $post['email'];
+            $hash = $this->mathRandom->getUniqueHash();
+
+            try {
+                $this->recommendation->setCustomerId($this->loggedCustomer->getId());
+                $this->recommendation->setEmail($email);
+                $this->recommendation->setHash($hash);
+                $this->recommendation->setStatus(false);
```

Comment:
Tworzenie rekomendacji powinno odbywać się w repository, controller powinien tylko wywołać odpowiednią funkcję z repository i cała reszta powinna dziać się właśnie tam - wygenerowanie hasha oraz ustawienie odpowiednich wartości i zapis do bazy. Funkcja ta powinna przyjmować tylko to, co niezbędne do utworzenia obiektu, czyli np. email z requestu czy ID customera. Możesz też w tej funkcji zwrócić hash, bo kilka linijek dalej potrzebyn jest do wygenerowania linku.

Wszystkie rzeczy związane z operacjami na modelach nie powinny znajdować się w controllerze, ponieważ kod jest wtedy bardziej przejrzysty oraz łatwiejszy do modyfikacji. Jeśli np. stwierdzimy, że nasze repository jest złe i musimy skorzystać z innego, wystarczy wtedy tylko podmienić preferencje w odpowiednim pliku xml i wszystko nadal będzie działać. Albo np. jeśli będzie potrzeba stworzenia rekomendacji z innego miejsca - nie trzeba będzie powielać tego samego kodu, wystarczy wtedy tylko wrzucić repository do konstruktora i korzystać ze stworzonych wcześniej funkcji.